lib/rules/define-props-declaration.js

```diff
@@ -27,20 +198,87 @@ module.exports = {
   },
   /** @param {RuleContext} context */
   create(context) {
+    const sourceCode = context.getSourceCode()
+
     const scriptSetup = utils.getScriptSetupElement(context)
     if (!scriptSetup || !utils.hasAttribute(scriptSetup, 'lang', 'ts')) {
       return {}
     }
 
     const defineType = context.options[0] || 'type-based'
+    const separateInterface = context.options[1]?.separateInterface || false
+
     return utils.defineScriptSetupVisitor(context, {
-      onDefinePropsEnter(node) {
+      onDefinePropsEnter(node, props) {
         switch (defineType) {
           case 'type-based': {
             if (node.arguments.length > 0) {
               context.report({
                 node,
-                messageId: 'hasArg'
+                messageId: 'hasArg',
+                *fix(fixer) {
```

Comment:
Please extract this function to the top scope to reduce indentation and thus make it easier to read.